src/ami-set.js

```diff
@@ -51,6 +55,49 @@ AmiSet.listAmiSets = async function () {
 };
 
 /**
+ * Checks if the AMIs from the amiSet are valid or not.
+ */
+
+async function checkAmi(ctx, region, ami, vtype) {
+  let request;
+  let missing = [];
+  let virtualizationType = '';
+
+  if (ami) {
+    request = {
+      ImageIds: [ami],
+    };
+    let exists = await amiExists(ctx.ec2[region], ami);
+    if (exists) {
+      virtualizationType = await getVirtualizationType(ctx.ec2[region], ami);
+      if (virtualizationType !== vtype) {
+        missing.push({imageId: ami, region: region, virtualizationType: virtualizationType});
+      }
+    } else {
+      missing.push({imageId: ami, region: region, virtualizationType: virtualizationType});
+    };
+  }
+  return missing;
+};
+
+AmiSet.validate = async function (ctx, amiSet) {
+  let missing = [];
+  let exists = false;
+
+  await Promise.all(amiSet.amis.map(async (def) => {
+
+    if (def.hvm) {
+      missing = missing.concat(await checkAmi(ctx, def.region, def.hvm, 'hvm'));
+    }
+    if (def.pv) {
+      missing = missing.concat(await checkAmi(ctx, def.region, def.pv, 'paravirtual'));
+    }
+  }));
+
+  return missing;
```

Comment:
I already return the vtype because I need to check that in the amiset tool: `missing = missing.concat(await checkAmi(ctx, def.region, def.hvm, 'hvm'));`
Now, adding the boolean `valid` on it too, it's much better indeed.